dataland-frontend/src/components/resources/frameworkDataSearch/TwoLayerDataTable.vue

```diff
@@ -226,23 +245,6 @@ export default defineComponent({
         },
       });
     },
-    /**
-     * Enables groupRowExpansion (and collapse) when clicking on the whole header row
-     * @param event a click event
-     */
-    expandRowGroupOnHeaderClick(event: Event) {
-      const id = (event.target as Element).id;
-
-      const matchingChild = Array.from((event.target as Element).children).filter((child: Element) =>
-        this.arrayOfKpiDataObjects.some((dataObject) => dataObject.subcategoryKey === child.id),
-      )[0];
-
-      if (matchingChild || this.arrayOfKpiDataObjects.some((dataObject) => dataObject.subcategoryKey === id)) {
-        const index = this.expandedRowGroups.indexOf(matchingChild?.id ?? id);
-        if (index === -1) this.expandedRowGroups.push(matchingChild?.id ?? id);
-        else this.expandedRowGroups.splice(index, 1);
-      }
-    },
```

Comment:
Just out of curiousity:
Why have we moved away from this approach, and instead implemented event listeners for clicks on rows?
Is there any advantage?   Was this broken before?

Reply:
The previous implementation added event listeners to every single html element on the page and did not remove them when leaving the page.

Another thing was that because we now use it on more than one framework we wanted to move the script to a component or utility in order to easily add to other framework views. Here the property `subcategoryKey` was being used and so we created a more generic code... which can be easily added to any framework view.

Reply:
Ah ok cool. Sounds great. Thx for the explanation 👍